server.go

```diff
@@ -777,6 +777,10 @@ func (s *server) Start() error {
 		srvrLog.Infof("Auto peer bootstrapping is disabled")
 	}
 
+	if !atomic.CompareAndSwapInt32(&s.started, 0, 1) {
```

Comment:
What's the reason for this change?

Reply:
This is needed in order to avoid a race between stop() and start() of the server (and other calls which assume that the server started when `s.started` is 1. 
The current code changes `s.started` from zero to one at the beginning of the start function. This opens the door to races.

By moving it to the end we can be sure that integration tests will not progress until the server actually completed the startup procedure.